backend/server.js

```diff
@@ -0,0 +1,356 @@
+  });
+});
+
+app.post('/signup', async (req, res) => {
+  const { username, password } = req.body;
+  try {
+    const salt = bcrypt.genSaltSync();
+    const strongPassword =
+      /^(?!.*\s)(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_₹]).{8,20}$/;
+
+    // checking if password match strongPassword = regex
+    if (password.match(strongPassword)) {
+      const newUser = await new User({
+        username,
+        password: bcrypt.hashSync(password, salt)
+      }).save();
+      res.status(201).json({
+        response: {
+          userId: newUser._id,
+          username: newUser.username,
+          accessToken: newUser.accessToken
+          // add storyCollection?
+        },
+        success: true
+      });
+    } else {
+      throw 'Password must contain at least 8 characters, at least one letter, one number and one special character';
+    }
+  } catch (error) {
+    res
+      .status(400)
+      .json({ response: 'problem with the signup', success: false });
+  }
+});
+
+// POST request for signing in, match username and password
+// if you include accessToken in your request = you are logged in
+app.post('/signin', async (req, res) => {
+  const { username, password } = req.body;
+
+  try {
+    const user = await User.findOne({ username });
+
+    if (user && bcrypt.compareSync(password, user.password)) {
+      res.status(200).json({
+        response: {
+          userId: user._id,
+          username: user.username,
+          accessToken: user.accessToken
+          // add storyCollection
+        },
+        success: true
+      });
+    } else {
+      res.status(404).json({
+        response: "Username or password doesn't match.",
+        success: false
+      });
+    }
+  } catch (error) {
+    res.status(400).json({ response: error, success: false });
+  }
+});
+
+// endpoint for images iported with cloudinary
+app.post('/storyimg', parser.single('image'), async (req, res) => {
+  try {
+    const storyimg = await new StoryImg({
+      name: req.body.filename,
+      imageUrl: req.file.path
+    }).save();
+    res.json(storyimg);
+  } catch (err) {
+    res.status(400).json({ errors: err.errors });
+  }
+});
+
+app.post('/storycollection', authenticateUser);
+app.post('/storycollection', async (req, res) => {
+  const { description, character } = req.body;
+  console.log(description, req.user);
+  try {
+    const newStoryCollection = await new StoryCollection({
+      description,
+      character
+    }).save();
+    const user = await User.findByIdAndUpdate(req.user._id, {
+      $push: {
+        storyCollection: newStoryCollection
+      }
+    });
+    res.status(201).json({ response: newStoryCollection, success: true });
+  } catch (error) {
+    res.status(400).json({ response: error, success: false });
+  }
+});
+
+// get bookshelf
+app.get('/storycollection/userid/:userid', async (req, res) => {
+  const { userId, storyCollectionId } = req.params;
```

Comment:
Hmm, this piece of code won't work I believe - there is no `storyCollectionId` passed as path param